test/unit/eigensolver/test_tridiag_solver_permutations.cpp

```diff
@@ -0,0 +1,198 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include "dlaf/eigensolver/tridiag_solver/permutations.h"
+
+#include "gtest/gtest.h"
+#include "dlaf_test/matrix/util_tile.h"
+#include "dlaf_test/util_types.h"
+
+using namespace dlaf;
+using namespace dlaf::test;
+using namespace dlaf::eigensolver::internal;
+
+template <typename Type>
+class TridiagEigensolverPermutationsTest : public ::testing::Test {};
+TYPED_TEST_SUITE(TridiagEigensolverPermutationsTest, RealMatrixElementTypes);
+
+// Initializes square input and output matices of size (n x n) and block size (nb x nb). The matrices are
+// discribed by a distribution and an array of input and output tiles respectively. Note that in contrast
+// to `Matrix<>`, the tiles are readily available, they are not stored in futures.
+//
+// The elements of the input matrix are as follows:
+//
+// `in_ij = i + j`
+//
+// The output matrix is set to zero.
+//
+template <class T>
+std::tuple<matrix::Distribution, std::vector<matrix::Tile<T, Device::CPU>>,
+           std::vector<matrix::Tile<T, Device::CPU>>>
```

Comment:
Just curious, is this a type that we would want to make more formal? Would it be possible to use `Matrix` here even though it's overkill?

Reply:
Perhaps we would indeed want to make this a formal type. A pair of `matrx::Distribution` and `std::vector<matrix::Tile<>>` represent a synchronous matrix as opposed to `Matrix<>` which stores futures and is an asynchronous matrix.

To use `Matrix<>`, future of tiles have to be first collected in a container such as `std::vector<future<Tile<>>>` and then unwrapped in a kernel passed to `pika::dataflow()` or `ex::when_all_vector()`.

Reply:
I don't know how useful this would be in other situations, so to avoid overdesigning this type would it be sufficient to just introduce something like:
```
template <typename T, Device D>
struct SyncMatrix {
  matrix::Distribution dist;
  std::vector<matrix::Tile<T, D>> tiles;
};
```
just to document the type at least a little bit? If it's useful elsewhere in the future we can further formalize it, but I at least dislike plain `tuple`s where a struct could be used instead.